### PR TITLE
Android 8 fix for getting GPS data when running in background

### PR DESCRIPTION
**Pull request by kurly1**
From Android 8 on, a app can only get frequently GPS data if runing in the foreground.

There for a FOREGROUND_SERVICE is added to be able to receive GPS data if the app is running in the background.

This PR is a extension of PR #30.

### CODE REVIEW

**Comment by kumy** [comment]
OK to test

**Comment by kumy** [comment]
retest this please

**Comment by bekuno** [comment]
retest this please

File: src/main/java/menion/android/whereyougo/utils/NotificationService.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    @Override
+    public IBinder onBind(Intent intent) {
+        // Used only in case if service are bound (Bound Services).
+        return null;
+    }
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        Logger.v(this.getClass().getName(), "onCreate()");
```
Comment (bekuno) [inline src/main/java/menion/android/whereyougo/utils/NotificationService.java]: Please replace this.getClass().getName() by a static definition.
Hardcoding the class name is ugly but the only reliable way to work with proguard.

File: src/main/java/menion/android/whereyougo/utils/NotificationService.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+public class NotificationService extends Service {
+    private static final int notification_id = 10;
+    private static final String NOTIFICATION_CHANNEL_ID = "menion.android.whereyougo.utils.NotificationService";
+    private boolean foreground = false;
+    private boolean runing = false;
+    private NotificationManager mNM;
+    private String contentTitel;
+
+    public static final String TITEL = "ContentTitel";
```
Comment (bekuno) [inline src/main/java/menion/android/whereyougo/utils/NotificationService.java]: Is TITEL visible to users?

**Comment by Lineflyer** [comment]
Please do not merge this yet, until we merged some bug fixes (which are currently targeted to the master branch) and merged to `release` once again.

**Comment by Lineflyer** [comment]
Open bugfix PRs have now been retargeted to `release`. So whenever ready this PR can be merged to `master` as we do not need to merge `master` to `release` for the needed bugfixes anymore. Removing label.

File: src/main/java/menion/android/whereyougo/utils/NotificationService.java (hunk 2 above)
Comment (kurly1) [reply src/main/java/menion/android/whereyougo/utils/NotificationService.java]: No, the TITEL constant is not visible to the user. It is ussed for the internal tagging of data.

File: src/main/java/menion/android/whereyougo/utils/NotificationService.java (hunk 1 above)
Comment (kurly1) [reply src/main/java/menion/android/whereyougo/utils/NotificationService.java]: I have replaced this by a static definition.

File: src/main/res/values-de/strings_pref.xml [hunk 3]
```diff
@@ -78,7 +78,7 @@
     <string name="pref_sensors_orient_filter_medium">Mittlerer Filter</string>
     <string name="pref_sensors_orient_filter_no_filter">Kein Filter</string>
     <string name="pref_statusbar">Statusbar-Symbol</string>
-    <string name="pref_statusbar_desc">Symbol in der Statusbar anzeigen</string>
+    <string name="pref_statusbar_desc">Symbol in der Statusbar anzeigen (Wen diese Option nicht verfügbar ist, wird die Statusbar durch eine andere Einstellung benötigt.)</string>
```
Comment (bekuno) [inline src/main/res/values-de/strings_pref.xml]: Better:
... (Wenn diese Option nicht verfügbar ist, so wird die Statusbar durch eine andere Einstellung verwendet)

File: src/main/res/values/strings_pref.xml [hunk 4]
```diff
@@ -84,7 +84,7 @@
     <string name="pref_sensors_orient_filter_medium">Medium filter</string>
     <string name="pref_sensors_orient_filter_no_filter">No filtration</string>
     <string name="pref_statusbar">Statusbar Icon</string>
-    <string name="pref_statusbar_desc">Show icon in statusbar</string>
+    <string name="pref_statusbar_desc">Show icon in statusbar (If this option is not available, the statusbar is forced by an other seting.)</string>
```
Comment (bekuno) [inline src/main/res/values/strings_pref.xml]: Better:
... (If this option is not available, the status bar is forced by another setting)

**Comment by bekuno** [comment]
Hello @kurly1 , can you finish here?

**Comment by kurly1** [comment]
Hello bekuno

Because of the corona virus I have currently no spare time to work on this PR. But I will get back to it as soon as possible.

**Comment by bekuno** [comment]
@kurly1 Ok, I wish you all the best in this special time.

File: src/main/res/values-de/strings_pref.xml (hunk 3 above)
Comment (kurly1) [reply src/main/res/values-de/strings_pref.xml:81]: Wold the following description be beter understandabele? Or would it be bether to remove this remark completly? 

EN: If this option cannot be selected, the status bar is forced by another setting.
GER: Wen diese Option nicht ausgewählt werden kann, wird die Statusbar durch eine andere Einstellung benötigt.

Comment (bekuno) [reply src/main/res/values-de/strings_pref.xml:81]: I personaly would leave them unchanged.

By the way, in the meantime the string translations are moved to crowdin.
Therefore no need to translate here.

Comment (kurly1) [reply src/main/res/values-de/strings_pref.xml]: Removed the braces and it's text.

File: src/main/res/values/strings_pref.xml (hunk 4 above)
Comment (kurly1) [reply src/main/res/values/strings_pref.xml]: Removed the braces and it's text.

**Comment by Lineflyer** [comment]
@kurly1 You should either remove the german translation completely or remember to approve it on Crowdin after this has been merged (+ 10min).

**Comment by kurly1** [comment]
@Lineflyer I’m not familiar with Crowdin and I don’t know how the german translation should be removed from this PR. Currently the last commit set’s the german translation back to the initial value.

Do you have any advice on how to remove this translation from the current PR?

**Comment by bekuno** [comment]
It's fine so.
No translation anymore in the PR.

**Comment by bekuno** [comment]
@kurly1 
Thanks for your work.

**Pull request merged**